### PR TITLE
Update wishlist cypress tests

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Updated cypress wishlist tests that were broken by latest wishlist updates.

## Related Issue

Closes PWA-1801

### Verification Steps

1. Go to the cypress folder on pwa studio
2. Run yarn test
3. Run both wishlist tests and there shouldn't be any errors.

## Test Plan

NA

#### Test scenario(s) for direct fix/feature
Example
-   Verify user is able to apply filters on category page
-   Verify user is able to apply filters on search page

#### Test scenario(s) for any existing impacted features/areas
Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

#### Test scenario(s) for any Magento Backend Supported Configurations
Example
- Update default Sort value in backend and repeat above scenarios

#### Is Browser/Device testing needed?
Example
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>

#### Any ad-hoc/edge case scenarios that need to be considered?
Example
- Apply all filters to get 0 results and then remove filters to see respective products 

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: venia-integration-tests/src/actions/cartPage/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import {
+    cartPageRoot,
+    kebabMenuButton,
+    saveForLaterButton
+} from '../../fields/cartPage';
+
+/**
+ * Utility to move a product from the cart page to a single wishlist
+ */
+export const moveProductFromCartToSigleWishlist = productName => {
```
Comment (supernova-at) [inline venia-integration-tests/src/actions/cartPage/index.js:10]:
```suggestion
export const moveProductFromCartToSingleWishlist = productName => {
```

File: venia-integration-tests/src/actions/productPage/index.js [hunk 2]
```diff
@@ -20,3 +21,17 @@ export const addProductToExistingWishlistFromDialog = wishlistName => {
         .contains(wishlistName)
         .click();
 };
+
+/**
+ * Utility function to add simple product to cart from the product page
+ *
+ * Note: This only works for simple product because it doesnt consider
+ * product options. A different funciton has to be used for different
```
Comment (supernova-at) [inline venia-integration-tests/src/actions/productPage/index.js:29]:
```suggestion
 * product options. A different function has to be used for different
```

I'm sorry I can't help it I was born this way 😭

File: venia-integration-tests/src/actions/wishlistPage/index.js [hunk 3]
```diff
@@ -19,3 +22,34 @@ export const createWishlist = wishlistName => {
...
+/**
+ * Utility function to remove given item from a single wishlist.
+ * The utility assumes that the action is called from the wishlist page.
+ *
+ * Note: Only useful if CE or EE with mutiple wishlists disabled
+ *
+ * @param {String} productName name of the product to remove
+ */
+export const removeProductFromSingleWishlist = productName => {
+    // assert Product Price exists
```
Comment (supernova-at) [inline venia-integration-tests/src/actions/wishlistPage/index.js:35]: Can drop this comment?

File: venia-integration-tests/src/assertions/wishlist/index.js [hunk 4]
```diff
@@ -26,19 +26,37 @@ export const assertProductInWishlist = productName => {
...
 
+/**
+ * Utility function to assert product does not exist in the wishlist
+ * @param {String} productName product to verify it does not exist in wishlist
+ */
+export const asserProductNotInWishlist = productName => {
+    // assert Product container exists
+    cy.get(wishlistRoot).should('exist');
+
+    // assert Product exists
```
Comment (supernova-at) [inline venia-integration-tests/src/assertions/wishlist/index.js:54]:
```suggestion
    // assert Product does not exist
```

File: venia-integration-tests/src/fixtures/productPage/index.js [hunk 5]
```diff
@@ -7,3 +7,11 @@ export const productValeriaTwoLayeredTank = {
     name: 'Valeria Two-Layer Tank',
     url: '/valeria-two-layer-tank.html'
 };
+
+/**
+ * Sample simple product
+ */
+export const silverAmorBandleSet = {
```
Comment (supernova-at) [inline venia-integration-tests/src/fixtures/productPage/index.js:14]:
```suggestion
export const silverAmorBangleSet = {
```

Will need to be updated where used as well

File: venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js [hunk 6]
```diff
@@ -25,19 +28,29 @@ const {
 } = accountAccessFixtures;
 const { wishlistPage } = myAccountMenuFixtures;
 const { categorySweaters, productCarinaCardigan } = categoryPageFixtures;
+const { cartPageRoute } = cartPageFixtures;
 const { homePage } = homePageFixtures;
 const { wishlistRoute } = wishlistFixtures;
-const { productValeriaTwoLayeredTank } = productPageFixtures;
+const {
+    productValeriaTwoLayeredTank,
+    silverAmorBandleSet
```
Comment (supernova-at) [inline venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js:36]:
```suggestion
    silverAmorBangleSet
```

File: venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js [hunk 7]
```diff
@@ -69,14 +82,24 @@ describe('verify single wishlist basic features', () => {
...
-        //This will be updated once https://jira.corp.magento.com/browse/PWA-1709 is code complete
         addProductToWishlistFromProductPage();
         cy.visitPage(wishlistRoute);
 
         assertProductInWishlist(productCarinaCardigan);
-        //This will be updated once https://jira.corp.magento.com/browse/PWA-1709 is code complete
-        // assertProductInWishlist(productValeriaTwoLayeredTank.url);
+        assertProductInWishlist(productValeriaTwoLayeredTank.name);
+
+        cy.visitPage(silverAmorBandleSet.url);
```
Comment (supernova-at) [inline venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js:91]:
```suggestion
        cy.visitPage(silverAmorBangleSet.url);
```

File: venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js [hunk 8]
```diff
@@ -69,14 +82,24 @@ describe('verify single wishlist basic features', () => {
...
+        addSimpleProductToCartFromProductPage();
+        cy.visitPage(cartPageRoute);
+        moveProductFromCartToSigleWishlist(silverAmorBandleSet.name);
```
Comment (supernova-at) [inline venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js:94]:
```suggestion
        moveProductFromCartToSigleWishlist(silverAmorBangleSet.name);
```

File: venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js [hunk 9]
```diff
@@ -69,14 +82,24 @@ describe('verify single wishlist basic features', () => {
...
+        cy.visitPage(wishlistRoute);
+
+        assertProductInWishlist(productCarinaCardigan);
+        assertProductInWishlist(productValeriaTwoLayeredTank.name);
+        assertProductInWishlist(silverAmorBandleSet.name);
```
Comment (supernova-at) [inline venia-integration-tests/src/tests/e2eTests/wishList/singleWishlistAddRemoveProduct.spec.js:99]:
```suggestion
        assertProductInWishlist(silverAmorBangleSet.name);
```

File: venia-integration-tests/src/actions/productPage/index.js (hunk 2 above)
Comment (revanth0212) [reply venia-integration-tests/src/actions/productPage/index.js:29]: Dude, thank you 🤣 I am laughing at myself.

File: venia-integration-tests/src/tests/integrationTests/wishList/verifyMultipleWishlistFeatures.spec.js [hunk 10]
```diff
@@ -218,13 +226,20 @@ describe('verify single wishlist basic features', () => {
...
         }).as('getWishlistConfig3');
         cy.visitPage(wishlistRoute);
-        cy.wait(['@getCustomerWishlist5']).its('response.body');
+        cy.wait(['@getCustomerWishlist6']).its('response.body');
         cy.wait(['@getWishlistConfig3']).its('response.body');
+        cy.wait(1000);
+        cy.intercept('GET', getNewCustomerWishlistCall, {
+            fixture:
+                'wishlist/multipleWishlist/wishlistExpandWishlistPageEarrings.json'
+        }).as('getCustomerWishlist8');
```
Comment (supernova-at) [inline venia-integration-tests/src/tests/integrationTests/wishList/verifyMultipleWishlistFeatures.spec.js:240]: What happened to `7` ?! 🙀 

😄

File: venia-integration-tests/src/fields/wishlist/index.js [hunk 11]
```diff
@@ -7,10 +7,15 @@ export const wishlistItemPrice = '*[class^="wishlistItem-priceContainer-"]';
 export const wishlistItemAddToCartButton =
     '*[class^="wishlistItem-addToCart-"]';
 
+export const wishlistItemRemoveButton = '*[class^="wishlistItem-delteItem-"]';
```
Comment (tjwiebell) [inline venia-integration-tests/src/fields/wishlist/index.js:10]: This selector is accurate, but maybe we should fix the typo in the components className.

```suggestion
export const wishlistItemRemoveButton = '*[class^="wishlistItem-deleteItem-"]';
```

File: venia-integration-tests/src/tests/integrationTests/wishList/verifyMultipleWishlistFeatures.spec.js (hunk 10 above)
Comment (revanth0212) [reply venia-integration-tests/src/tests/integrationTests/wishList/verifyMultipleWishlistFeatures.spec.js]: 🤣 

Fixed

File: venia-integration-tests/src/fields/wishlist/index.js (hunk 11 above)
Comment (revanth0212) [reply venia-integration-tests/src/fields/wishlist/index.js]: You are right, i have fixed it in 46bc993

**Review by supernova-at — APPROVED** [review]

**Comment by supernova-at** [comment]
✅  Code looks good
❓ Haven't been able to actually verify the tests yet

**Comment by supernova-at** [comment]
On verification, I think we missed a typo somewhere - the actual element is `delteItem` and the Cypress test is looking for `deleteItem`:

Screen Shot 2021-07-13 at 1 38 40 PM

Screen Shot 2021-07-13 at 1 38 54 PM

**Comment by revanth0212** [comment]
Ohh yeah we might have to update develop for that but if you test this against a local instance or https://pr-3257.pwa-venia.com/ it should work because I updated the component's CSS class to `deleteItem`

**Comment by supernova-at** [comment]
Oops yep, my bad.

✅  Verification steps passed!
